specs/DesignDiscussions/UITableView.md

```diff
@@ -0,0 +1,118 @@
+# UITableView
+## Problem Description
+Using UITableView as a hosting view for AdaptiveCard requires additional considerations. One of the challenges is the height changes in the row. UITableView correctly self sizes most of the times, but it breaks down in a few known cases, for example, if the images are loaded asynchronously causing height changes after the row is drawn, or if the visibility changes in its subviews, the row does not adjust the height dynamically. In such cases, there are a few well known practices in dealing with such occasions. This document covers practical guides in dealing with the height changes. This document will focus on [beginUpdates()](https://developer.apple.com/documentation/uikit/uitableview/1614908-beginupdates) and [reloadRows(at:with:)](https://developer.apple.com/documentation/uikit/uitableview/1614935-reloadrows) methods. It will show how to use the APIs in conjunction with AdaptiveCards' API. The rest of the documents uses the [iOS AdaptiveCard sample app](https://github.com/microsoft/AdaptiveCards/tree/main/source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizer) as an example.
```

Comment:
`UITableView correctly self sizes most of the times, but it breaks down in a few known cases, for example, if the images are loaded asynchronously causing height changes after the row is drawn, or if the visibility changes in its subviews, the row does not adjust the height dynamically.`

This sentence is a bit difficult to read and super long...can you rephrase it a little bit?

Reply:
sure; thank you for the feedback.